azurelinuxagent/common/protocol/metadata.py

```diff
@@ -25,9 +25,14 @@
 import traceback
 
 import azurelinuxagent.common.conf as conf
+from azurelinuxagent.common.datacontract import get_properties, set_properties, validate_param
+from azurelinuxagent.common.exception import HttpError, ProtocolError
```

Comment:
what was happening here? I see metadata.py does reference HttpError, etc... but there was no import before???

Reply:
was working because of the import * from restapi at line 38?

Reply:
restapi was importing this `from azurelinuxagent.common.exception import ProtocolError, HttpError` before and the `import *` was pulling it in

Reply:
yup --- hidden/unintended dependencies --- please stay away from this kind of imports in new code (and let's try to fix existing code when possible)

Reply:
Exactly, my reasoning was to eliminate all the `*` imports in the files in was changing. It's scary.